### PR TITLE
Density - various object Pt 1

### PR DESCRIPTION
**Pull request by batekulve**
Jewelry, fuel, electric motors, vehicle batteries, bedding, mutagen, tool_armors, clown nose

#### Summary
Balance "Density balance pass for jewelry, fuel, electric motors, vehicle batteries, bedding, mutagen, tool_armors, and clown nose."

#### Purpose of change
Fixing some objects' density being either impossibly high, or in some cases unreasonably low. Contributes toward fixing #54811. Re-done version of pull request #57105

#### Describe the solution
If an object's density was higher than its materials suggests, then lower it to the correct density. If an object's density was significantly lower than its material suggests and below 1.0 g/cm3, then raise it, but not all the way to the cap.

#### Describe alternatives you've considered
In real life, lithium-ion batteries are less dense than lead-acid batteries. However, Li+ storage batteries in Cataclysm can hold _way_ more charge than real life Li+ vehicle batteries. I think having storage batteries be about twice as heavy as lead-acid car batteries is a good balance between realism and game balance. This is still lighter then they were before.

I could have left watch volume the way it was, but I think 250 ml (the volume of a plastic water bottle) was way too high for even a large watch. Online stats for Rolex watches put the weight at somewhere between 83 to 225 g.

#### Testing
Copied item JSON to latest experimental game directory. Weights and volumes in the item spawning menu were correct.

#### Additional context
Enhanced electric motors are made of superalloy. Are the more advanced Super electric motors supposed to be made out of superalloy as well?

### CODE REVIEW

**Comment by batekulve** [comment]
The number of files changed being different is the reason I ran into conflicts in the previous pull request. The emergency blanket was moved to a different file. Also I fixed the clown nose.

**Comment by esotericist** [comment]
if you're going to be tinkering with balance-critical things (particularly things the tests make assertions about), please provide citations for where you got your numbers from.

**Comment by kevingranade** [comment]
>  I think having storage batteries be about twice as heavy as lead-acid car batteries is a good balance between realism and game balance.

This is an invalid rationale, they need to reflect real world batteries, that is all.

**Comment by Drew4484** [comment]
Don't change numbers for game balance, change them for accuracy. If storage batteries have been underweight, players will need to review their vehicle decisions. 

An electric motor does not need superalloy, aluminum is more appropriate if you're going to be changing things.

**Comment by esotericist** [comment]
so, looking again, it seems like the core problem is that this PR is only attempting to make the density test pass, without regard for how it comes around to passing.

the purpose of a test like this is to recognize a data consistency error exists, but not dictate how the consistency error should be addressed. most notably, there is a generalized expectation that if something is failing a test, anyone seeking to fix the thing should be looking for *what is most correct*, not merely what causes the test to stop complaining about the thing.

as a generalization (for non-specific values of 'you', this isn't directed solely at the PR author here): if you cannot find (or cannot provide meaningful citations on request) the correct values to be used for a given item, you shouldn't be tinkering with it.

and we're pretty much always going to ask for citations for things that have cascading game mechanic implications (and most especially stuff that is directly or indirectly evaluated by multiple tests, such as weapons, armor, or vehicle parts).

**Comment by batekulve** [comment]
The thing about those density tests - as I understand it - is that it fails when an object's density is more dense than if the entire thing was made of its most dense material. For example, a pair of copper earrings has a volume of 1 ml and used to weigh 50 grams, which makes it more dense than any non-fictional material in materials.json. This, obviously, doesn't make sense, and given the choice between dropping the weight or increasing the volume of copper earrings, I dropped the weight. In almost all instances of failures like this I drop the weight instead of increasing the volume.

In other words, the purpose of the test is to find objects whose density needs to be lowered, and I'm doing that not to make the tests pass but for those objects to make physical sense.

Storage batteries haven't been underweight, they've been massively overweight. Almost five times overweight. However, they are also much more energy dense than real life lithium-ion batteries. Which is why, instead of lowering them all the way down to real-life Li+ battery density, I only lowered them slightly to just under the density of ceramic. Their heaviness is really the only drawback to storage batteries. If they weighed even less than car batteries they would be vastly superior. It's fine if you're okay with that, of course, but I thought that sounded too overpowered.

**Comment by kevingranade** [comment]
> If they weighed even less than car batteries they would be vastly superior. It's fine if you're okay with that, of course, but I thought that sounded too overpowered.

If you have a well sourced insight into weightvdensityvand charge density, fix both things.

**Comment by batekulve** [comment]
> > If they weighed even less than car batteries they would be vastly superior. It's fine if you're okay with that, of course, but I thought that sounded too overpowered.
> 
> If you have a well sourced insight into weight density and charge density, fix both things.

I can do that. Is it okay to use this pull request?

**Comment by Zireael07** [comment]
@batekulve I'd make a separate PR for charge/weight

**Comment by batekulve** [comment]
That pull request is in progress. Any other specific feedback?

File: data/json/items/vehicle/motors.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "id": "motor_enhanced",
     "name": { "str": "enhanced electric motor" },
     "description": "A very powerful and yet lightweight electric motor.  Useful for crafting.",
-    "weight": "100 kg",
+    "weight": "35 kg",
```
Comment (kevingranade) [inline data/json/items/vehicle/motors.json:22]: This is no bueno possibly this should have its materials adjusted, maybe add copper.
```suggestion
    "weight": "100 kg",
```

File: tests/vehicle_drag_test.cpp [hunk 2]
```diff
@@ -244,8 +244,8 @@ TEST_CASE( "vehicle_drag", "[vehicle] [engine]" )
     test_vehicle_drag( "car", 0.294604, 2.255646, 1064.506250, 11995, 14426 );
     test_vehicle_drag( "car_mini", 0.294604, 1.599943, 1132.590625, 12238, 14657 );
     test_vehicle_drag( "car_sports", 0.294604, 2.331567, 1320.402500, 20918, 24972 );
-    test_vehicle_drag( "car_sports_atomic", 0.294604, 3.185611, 1503.384375, 23795, 24693 );
-    test_vehicle_drag( "car_sports_electric", 0.294604, 1.798178, 1018.336250, 23968, 24864 );
+    test_vehicle_drag( "car_sports_atomic", 0.294604, 2.898668, 1367.967708, 23795, 24693 );
```
Comment (kevingranade) [inline tests/vehicle_drag_test.cpp:247]:
```suggestion
    test_vehicle_drag( "car_sports_atomic", 0.294604, 3.185611, 1503.384375, 23795, 24693 );
```

File: tests/vehicle_drag_test.cpp [hunk 3]
```diff
@@ -244,8 +244,8 @@ TEST_CASE( "vehicle_drag", "[vehicle] [engine]" )
...
+    test_vehicle_drag( "car_sports_electric", 0.294604, 1.511235, 855.836250, 23968, 24864 );
```
Comment (kevingranade) [inline tests/vehicle_drag_test.cpp:248]:
```suggestion
    test_vehicle_drag( "car_sports_electric", 0.294604, 1.798178, 1018.336250, 23968, 24864 );
```

**Comment by batekulve** [comment]
@kevingranade I've implemented your solutions

**Comment by Night-Pryanik** [comment]
Could you please resolve conflicts?

**Comment by batekulve** [comment]
Conflicts resolved. Now it's saying the density tests are required and failing. Anything I need to do about that?

**Comment by Night-Pryanik** [comment]
Well, since you're messing with density stuff, you could take a look at why the density tests are falling.

**Comment by batekulve** [comment]
I'm not sure what that means but it doesn't look related to my changes.

**Comment by anothersimulacrum** [comment]
It doesn't look so to me either.

File: data/json/items/armor/jewelry.json [hunk 4]
```diff
@@ -325,7 +325,7 @@
     "type": "ARMOR",
     "name": { "str": "copper hairpin" },
     "description": "A hairpin made from copper.",
-    "weight": "2 g",
+    "weight": "18 g",
```
Comment (Maleclypse) [inline data/json/items/armor/jewelry.json:328]: 18 grams seems high for 2ml?

**Review by Maleclypse — APPROVED** [review]
Thanks for doing this!

**Pull request merged**